### ISSUE TITLE
Go to definition id opening file in lowercase

### ISSUE THREAD

**Issue by babobski** [Komodo/KomodoEdit#3353]
### Short Summary
I'm currently working on a addon for mapping remote to local.
Only I noticed the Go to dfinition is not working correct.

If I have a remote project I always map the php to my local copy of the project.
Only when I use the Go to definition, the file is openend wrong.

Everything is converted to lowercase.

### Steps to Reproduce
Use the go to definition.

### Expected results
The file is opend with the correct file name and path/uri.

### Actual results
![afbeelding](https://user-images.githubusercontent.com/3530262/32658326-b98f8112-c619-11e7-8236-8d40727e7dbf.png)

If i know copy the path it is also incorrect:

```
c:\...\vacantievreugd dev\vendor\....\components\eomailer.php
```

It should be:
```
C:\...\vacantievreugd DEV\vendor\....\components\EOMailer.php
```

### Platform Information

Komodo IDE, version 10.2.3, build 89902, platform win32-x86

Do you guys have a idea how I can get the correct url from this maleformed url

**Comment by mitchell-as** [comment]
This is just how CodeIntel works -- it normalizes Windows filenames to be all lower-case since the Windows filesystem is case-insensitive. It should still be a valid name on the remote filesystem, no?

**Comment by Naatan** [comment]
This has nothing to do with CodeIntel, it's file opening and the UI. @cgchoffman this sounds like the issue you were looking into a while ago, did we have a bug open for that?

Ie. the issue where Komodo is calling abspath way too generously.

**Comment by cgchoffman** [comment]
I hit an issue like this but was never able to reproduce it.

**Comment by babobski** [comment]
It's still a valid url, it opens the file. Only the display of the filename is a bit weird, the file is `EOHelper` and not `eohelper`. I am using the project url to see if the local file is on the project folder, so I can open the mirrored remote file.

Only when using it with the go to definition, I got a miss match of my urls because the file path has become lowercase. So as workaround I should do the testing in lowercase, I think that should work in my case.

**Comment by mitchell-as** [comment]
Yes @babobski, for case-insensitive filesystems, you should always normalize filenames before comparing them.

**Comment by Naatan** [comment]
Worth noting there's a distinction between normalizing paths for comparisons and just plain normalizing paths. At the moment Komodo is doing to much of the latter.

**Comment by babobski** [comment]
Thanks, would the best way to compare the two urls, is to use the `window.encodeURIComponent`?
Still hitting some issue's with some urls, only converting them to lowercase did not do the trick.

**Comment by babobski** [comment]
Note, this is a serious issue when combined with publishing.
When I use the go to definition to go to a file, make some changes.
This will result in two files on the remote server, also you don't see you're changes as you're making the changes in a "other" file
